src/Disks/IO/WriteBufferFromAzureBlobStorage.cpp

```diff
@@ -21,18 +21,30 @@ namespace DB
 WriteBufferFromAzureBlobStorage::WriteBufferFromAzureBlobStorage(
     std::shared_ptr<const Azure::Storage::Blobs::BlobContainerClient> blob_container_client_,
     const String & blob_path_,
-    size_t max_single_part_upload_size_,
-    size_t max_unexpected_write_error_retries_,
     size_t buf_size_,
-    const WriteSettings & write_settings_)
+    const WriteSettings & write_settings_,
+    std::shared_ptr<const AzureObjectStorageSettings> settings_,
+    ThreadPoolCallbackRunner<void> schedule_)
     : WriteBufferFromFileBase(buf_size_, nullptr, 0)
     , log(getLogger("WriteBufferFromAzureBlobStorage"))
-    , max_single_part_upload_size(max_single_part_upload_size_)
-    , max_unexpected_write_error_retries(max_unexpected_write_error_retries_)
+    , buffer_allocation_policy(IBufferAllocationPolicy::create({settings_->strict_upload_part_size,
+                                                                settings_->min_upload_part_size,
+                                                                settings_->max_upload_part_size,
+                                                                settings_->upload_part_size_multiply_factor,
+                                                                settings_->upload_part_size_multiply_parts_count_threshold,
+                                                                settings_->max_single_part_upload_size}))
```

Comment:
This code is difficult to check to be sure that you assign fields of `BufferAllocationPolity::Settings` to the matching fields of `AzureObjectStorageSettings`. Just a simple helper function could make that easier:

```
namespace
{
    BufferAllocationPolicyPtr createBufferAllocationPolicy(const AzureObjectStorageSettings & settings)
    {
        BufferAllocationPolicy::Settings alloc_settings;
        alloc_settings.strict_size = settings.strict_upload_part_size;
        ...
        return BufferAllocationPolicy::create(alloc_settings);
    }
}

WriteBufferFromAzureBlobStorage::WriteBufferFromAzureBlobStorage(... std::shared_ptr<const AzureObjectStorageSettings> settings_, ...)
    : buffer_allocation_policy(createBufferAllocationPolicy(*settings_), ...
```

Reply:
ya good point, addressed it